app/controllers/messages_controller.rb

```diff
@@ -52,6 +48,25 @@ def destroy
   def post
     authorize @message
     @message.update_attributes(posted: true)
+
+    devices = Device.all
+    client = Exponent::Push::Client.new
+    messages = []
```

Comment:
Hey Narayan. Typically when you have array in this case `devices` and from that array you want to create another array in this case `messages`. You should use the map method. So the code would be:
```ruby
messages = devices.map do |device|
  {
    to: device.token,
    sound: "default",
    body: @message.title
  }
end
```